src/main/java/com/google/cloud/tools/appengine/cloudsdk/CloudSdkAppEngineDevServer1.java

```diff
@@ -112,8 +115,11 @@ public void run(RunConfiguration config) throws AppEngineException {
     for (File service : config.getServices()) {
       arguments.add(service.toPath().toString());
     }
+
+    Map<String, String> appEngineEnvironment = getAllAppEngineEnvironment(config.getServices());
```

Comment:
I'm not sure about this. If you look at the isJava8 method in CloudSdkAppEngineDevServer1 its doing something similar (parsing the appengine-web.xml, and it does this every time).

I think of this preparation of the env vars from appengine-web.xml as similar to the preparation of the arguments to the command (which happens in this class).